2-mandatory/3-financial-times.js

```diff
@@ -14,24 +20,52 @@ function potentialHeadlines(allArticleTitles) {
     (you can assume words will always be seperated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+
+    const lengthTitles = [];
+    let temp = 0;
+    for (let allTitles of allArticleTitles) {
+        temp = allTitles.split(" ").length;
+        lengthTitles.push(temp);
+    }
+
+    lengthTitles.sort(function (a, b) { return a - b });
+
+    for (let allTitles of allArticleTitles) {
+        temp = allTitles.split(" ").length;
+        if (temp === lengthTitles[0])
+            return allTitles;
+    }
+
 }
 
+
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+
+    let numberIncluded = [];
+    for (let allTitles of allArticleTitles)
+        if (allTitles.includes(0 || 1 || 2 || 3 || 4 || 5 || 6 || 7 || 8 || 9))
```

Comment:
having lots of things in a conditional can be unmaintainable in the long run, what you could do here is create a separate array which contains the numbers 0-9 (`const validNums = [...Array(10).keys()]`) and then create a separate function which returns this conditional:

```
const validNums = [...Array(10).keys()]

function arrIncludesValidNum(arr) {
   return validNums.some(element => arr.includes(element))
}

...

if(arrIncludesValidNum(allTitles)) {
   ...
}
```
This is way more advanced than is necessary here, however it's a way to make your code more readable/maintainable in the long run.

Reply:
Thank you. Your answer is amazing. I couldn't understand  [...Array(10).keys()]